### PR TITLE
Allow RPED to upgrade APC power cells.

### LINKED ISSUE #33804: Feature request: Rapid cell swapper
[featurerequest]

RTEG cells are actually usable now but trying to upgrade APCs makes my soul scream after the first 2.
I think there should be a high-end gadget similar to the rapid part exchange device but for APCs

### PR DESCRIPTION
**Pull request by twirling-jesters**
closes #33804

## What this does
Allows RPEDs to be used on APCs, swapping out the APC power cell for the highest grade power cell in the RPED.

## Why it's good
Makes upgrading the station's APCs a much more streamlined process.

## Changelog
:cl:
 * rscadd: RPEDs can now be used to upgrade APCs.

### CODE REVIEW

**Comment by Eneocho** [comment]
Whilst very helpful, this is kinda inconsistent, since the RPED still needs the device to be screwed open to upgrade while the BRPED doesn't.
I'd make it so RPEDs can upgrade the APC if you unscrew first (instead of having to unlock, then UI, then crowbar), whilst BRPEDs can swap with no extra requirement.

**Comment by Eneocho** [comment]
I'd implement a priority system for the cells, since just using the highest charge to determine which cell is better will cause issues with the self-charging RTG and PDTG cells, which have 1500w and 2500w respectively. A solution to that would be making the selfcharging cells have higher priority than regular cells. This would ensure the Infinite (adminbus), PDTG and RTG cells go first, then the Ultra, Hyper, Super, High and regular powercells.

**Comment by twirling-jesters** [comment]
Fair points, I went ahead and added some ratings to the cells so that the RPED can discern which ones to replace and added the APC being opened/wires exposed as a condition to exchange the power cell with a standard RPED.

**Comment by Aloraydrel** [comment]
Soulless...

**Comment by SECBATON-GRIFFON** [comment]
i've been thinking of a way to do this with robotic components too

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -660,6 +662,25 @@
 				"You hear bang")*/
 			..() //Sanity
 
+/obj/machinery/power/apc/exchange_parts(mob/user, obj/item/weapon/storage/bag/gadgets/part_replacer/W)
+	if (W.bluespace || wiresexposed || opened)
+		var/obj/item/weapon/cell/max_cell
+		for(var/obj/item/weapon/cell/component in W.contents)
+			if (!max_cell || max_cell.rating < component.rating)
```
Comment (Kurfursten) [inline code/modules/power/apc.dm]: place the right side of the or statement in parenthesis to be safe, I think this might interpret the || before the <

**Pull request merged**